### PR TITLE
LOOP-3644: Additional instructional content in the Notifications permission priming screen

### PR DESCRIPTION
**Pull request by rickpasetto**
https://tidepool.atlassian.net/browse/LOOP-3644

### CODE REVIEW

File: TidepoolOnboarding/Assets.xcassets/Notification_1.imageset/Contents.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "scale" : "1x"
+    },
+    {
+      "filename" : "notification_1.pdf",
```
Comment (darinkrauss) [inline TidepoolOnboarding/Assets.xcassets/Notification_1.imageset/Contents.json]: The filename should match the asset name, specifically "Notification_1" (note capitalization).

File: TidepoolOnboarding/Views/Accessory/Callout.swift [hunk 2]
```diff
@@ -40,27 +42,12 @@ struct Callout<Content: View>: View {
         .overlay(RoundedRectangle(cornerRadius: 10, style: .continuous).stroke(Color(.systemFill), lineWidth: 1))
         .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
     }
-
-    private var warning: some View {
```
Comment (darinkrauss) [inline TidepoolOnboarding/Views/Accessory/Callout.swift:44]: Was this extra code?

File: TidepoolOnboarding/Views/Sections/YourDevicesViews.swift [hunk 3]
```diff
@@ -56,8 +57,15 @@ fileprivate struct YourDevicesNotificationsView: View {
                 LocalizedString("Enable Critical Alerts in your iPhone or iPod Touch Settings", comment: "Onboarding, Your Devices section, Notifications view, list 2, item 1")
             )
             .startingAt(2)
-            .padding(.vertical)
+            .padding(.top)
             Paragraph(LocalizedString("Critical Alerts may be configured to alert you to higher risk situations while using Tidepool Loop, such as urgent low glucose, insulin pump occlusions, or other serious system errors.", comment: "Onboarding, Your Devices section, Notifications view, paragraph 3"))
+            .padding(.vertical)
+            Callout(title: LocalizedString("Notifications and Critical Alert permissions must be allowed to continue using the app", comment: "Onboarding, Your Devices section, Notifications view, callout title"),
+                    warningIconColor: .red) {
```
Comment (darinkrauss) [inline TidepoolOnboarding/Views/Sections/YourDevicesViews.swift:62]: [nit] Seems silly to me to have two different warning colors.

File: TidepoolOnboarding/Views/Sections/YourDevicesViews.swift [hunk 4]
```diff
@@ -56,8 +57,15 @@ fileprivate struct YourDevicesNotificationsView: View {
                 LocalizedString("Enable Critical Alerts in your iPhone or iPod Touch Settings", comment: "Onboarding, Your Devices section, Notifications view, list 2, item 1")
             )
             .startingAt(2)
-            .padding(.vertical)
+            .padding(.top)
             Paragraph(LocalizedString("Critical Alerts may be configured to alert you to higher risk situations while using Tidepool Loop, such as urgent low glucose, insulin pump occlusions, or other serious system errors.", comment: "Onboarding, Your Devices section, Notifications view, paragraph 3"))
+            .padding(.vertical)
```
Comment (darinkrauss) [inline TidepoolOnboarding/Views/Sections/YourDevicesViews.swift]: [nit] Please indent.

File: TidepoolOnboarding/Views/Sections/YourDevicesViews.swift [hunk 5]
```diff
@@ -56,8 +57,15 @@ fileprivate struct YourDevicesNotificationsView: View {
...
+            Callout(title: LocalizedString("Notifications and Critical Alert permissions must be allowed to continue using the app", comment: "Onboarding, Your Devices section, Notifications view, callout title"),
+                    warningIconColor: .red) {
+                Paragraph(LocalizedString("It is important that you always keep Notifications and Critical Alerts turned on in your phone’s settings to ensure that you receive Tidepool Loop notifications.", comment: "Onboarding, Your Devices section, Notifications view, callout body text"))
+            }
+            Paragraph(LocalizedString("Additional preferences can be set within the device manager screens of the Tidepool Loop app.", comment: "Onboarding, Your Devices section, Notifications view, paragraph 4"))
+            .padding(.vertical)
```
Comment (darinkrauss) [inline TidepoolOnboarding/Views/Sections/YourDevicesViews.swift]: [nit] Please indent.

**Review by darinkrauss — CHANGES_REQUESTED** [review]
A few minor changes.

File: TidepoolOnboarding/Views/Accessory/Callout.swift (hunk 2 above)
Comment (rickpasetto) [reply TidepoolOnboarding/Views/Accessory/Callout.swift:44]: Yes, this private `var` was never used, so I removed it.  I'm _guessing_ it was replaced by `WarningIcon` when it was created.

File: TidepoolOnboarding/Views/Sections/YourDevicesViews.swift (hunk 3 above)
Comment (rickpasetto) [reply TidepoolOnboarding/Views/Sections/YourDevicesViews.swift:62]: I'm just following the [design spec](https://www.figma.com/file/JFIqEtYArG0oMWJCgLbQL5/LOOP-3644-3645?node-id=201%3A2720). 🤷‍♂️

File: TidepoolOnboarding/Views/Sections/YourDevicesViews.swift (hunk 4 above)
Comment (rickpasetto) [reply TidepoolOnboarding/Views/Sections/YourDevicesViews.swift]: I'll try to follow the whitespace/indent pattern in the rest of the file.

File: TidepoolOnboarding/Views/Sections/YourDevicesViews.swift (hunk 5 above)
Comment (rickpasetto) [reply TidepoolOnboarding/Views/Sections/YourDevicesViews.swift]: I'll try to follow the whitespace/indent pattern in the rest of the file.

File: TidepoolOnboarding/Views/Sections/YourDevicesViews.swift (hunk 4 above)
Comment (darinkrauss) [reply TidepoolOnboarding/Views/Sections/YourDevicesViews.swift]: Ah, yes, you pulled the closing paren for the `Paragraph` down to the next line. The rest of the code has it on the same line and then the `.padding(...)` would be indented automatically.

File: TidepoolOnboarding/Views/Sections/YourDevicesViews.swift (hunk 5 above)
Comment (darinkrauss) [reply TidepoolOnboarding/Views/Sections/YourDevicesViews.swift]: Same as above.

File: TidepoolOnboarding/Views/Accessory/Callout.swift (hunk 2 above)
Comment (darinkrauss) [reply TidepoolOnboarding/Views/Accessory/Callout.swift:44]: Cool, I must have missed that when I refactored `WarningIcon`.

File: TidepoolOnboarding/Views/Sections/YourDevicesViews.swift (hunk 3 above)
Comment (darinkrauss) [reply TidepoolOnboarding/Views/Sections/YourDevicesViews.swift:62]: :+1:

**Review by darinkrauss — APPROVED** [review]
Approved, but would appreciate fixing the two trailing parens and the following indented lines. Thanks!

**Pull request merged**